hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/PrisonApiService.kt

```diff
@@ -1,38 +1,30 @@
 package uk.gov.justice.digital.hmpps.prisonersearch.search.services
 
-import kotlinx.coroutines.async
-import kotlinx.coroutines.reactive.awaitSingle
-import kotlinx.coroutines.runBlocking
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.bodyToMono
 
 @Component
 class PrisonApiService(val prisonApiWebClient: WebClient) {
 
-  fun getAllAlerts(): List<AlertType> {
-    return runBlocking {
-      val alertTypes = async { getDomainReferenceCodes("ALERT") }
-      val alertCodes = async { getDomainReferenceCodes("ALERT_CODE") }
-
-      alertTypes.await().map { alertType ->
-        AlertType(
-          type = alertType.code,
-          description = alertType.description,
-          active = alertType.activeFlag == "Y",
-          alertCodes = alertCodes.await()
-            .filter { it.parentCode == alertType.code }
-            .map { AlertCode(it.code, it.description, it.activeFlag == "Y") },
-        )
-      }
+  fun getAllAlerts(): List<AlertType> =
+    getAlertTypes().map { type ->
+      AlertType(
+        type = type.code,
+        description = type.description,
+        active = type.activeFlag == "Y",
+        alertCodes = type.subCodes.map {
+          AlertCode(it.code, it.description, it.activeFlag == "Y")
+        },
+      )
     }
-  }
 
-  private suspend fun getDomainReferenceCodes(domain: String) = prisonApiWebClient.get()
-    .uri("/api/reference-domains/domains/{domain}/codes", domain)
+  private fun getAlertTypes() = prisonApiWebClient.get()
+    .uri("/api/reference-domains/alertTypes")
+    .header("Page-Limit", "1000")
```

Comment:
We have 20 different alert types in prod but in an attack of over-cautiousness I went big.